packages/components/src/toolbar/stories/index.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import Toolbar from '../';
+import { SVG, Path, ToolbarButton, ToolbarGroup } from '../../';
+
+export default { title: 'Components|Toolbar', component: Toolbar };
+
+function InlineImageIcon() {
+	return (
+		<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+			<Path d="M4 16h10c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v9c0 1.1.9 2 2 2zM4 5h10v9H4V5zm14 9v2h4v-2h-4zM2 20h20v-2H2v2zm6.4-8.8L7 9.4 5 12h8l-2.6-3.4-2 2.6z" />
+		</SVG>
+	);
+}
+
+const ToolbarWithId = withInstanceId( ( { instanceId, ...props } ) => (
```

Comment:
Reakit generates random IDs for each roving tabindex item. By default, they'll be different between client, server and snapshots (in this case, storyshots). If we pass `id="toolbar"` to `Toolbar`, that's guaranteed that they will be `toolbar-1`, `toolbar-2` etc.

The module that makes it possible underneath is still not public, but more information can be found here (this page is still not available in the navigation): https://reakit.io/docs/id/

This is only necessary for snapshots and server side rendering. In those cases, users can do `<Toolbar id="...">`. Do you think it's necessary to add `withInstanceId` to the toolbar implementation? I've used it on the snapshots because eslint complained when I passed an `id` directly.

Reply:
Those stories are going to be exposed in Storybook and people might think that you have to use `id` with the current setup. It turns out that it isn't mandatory unless you want to support server side rendering. I don't know what is the best strategy to move forward. Based on your comments, I don't think we should enforce injecting `id` on all toolbars.

Maybe we should disable ESLint rules and provide inline comments explaining why it has to work this way? I don't think it's perfect as well. In the current shape, I'm sure some people will copy `ToolbarWithId` part as well :)

Reply:
Yeah! I agree with you! Will disable the eslint rule for this file then.